http2/transport_test.go

```diff
@@ -5425,6 +5425,49 @@ func TestIssue67671(t *testing.T) {
 	}
 }
 
+// Issue 70809: when StrictMaxConcurrentStreams enabled ClientConn.ReserveNewRequest() causes stalls in request processing
+func TestIssue70809(t *testing.T) {
+	ts := newTestServer(t, func(w http.ResponseWriter, r *http.Request) {}, func(s *Server) {
+		s.MaxConcurrentStreams = 1
+	})
+	tr := &Transport{
+		TLSClientConfig:            tlsConfigInsecure,
+		AllowHTTP:                  true,
+		StrictMaxConcurrentStreams: true,
+	}
+
+	var opt RoundTripOpt
+
+	get := func() int {
+		req, _ := http.NewRequest("GET", ts.URL, nil)
+		res, err := tr.RoundTripOpt(req, opt)
+		if err != nil {
+			t.Error(err)
+		}
+		res.Body.Close()
+		return res.StatusCode
+	}
```

Comment:
How about adding a timeout for the test to eventually fail? Right now it hangs forever.